# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvas.cpp
# Commit: 904012057423
# Full Hash: 904012057423b23bf9a382da27a40869c6c00301
# Author: Ehsan Akhgari <ehsan@mozilla.com>
# Date: 2019-03-06 09:56:17
# Regressor Bug: 1532414
# File Overlap Count: 1
# Description:
#   Bug 1532414 - Pass a document/principal pointer to nsContentUtils::ShouldResistFingerprinting() callers in the canvas API; r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22002
# ==============================================================================

diff -r 152b9d59cb29 -r 904012057423 dom/canvas/OffscreenCanvas.cpp
--- a/dom/canvas/OffscreenCanvas.cpp	Tue Mar 05 23:13:23 2019 +0000
+++ b/dom/canvas/OffscreenCanvas.cpp	Tue Mar 05 23:20:08 2019 +0000
@@ -255,7 +255,9 @@
   // so that we can decide when usePlaceholder should be true/false?
   // See https://trac.torproject.org/18599
   // For now, we always return a placeholder if fingerprinting resistance is on.
-  bool usePlaceholder = nsContentUtils::ShouldResistFingerprinting();
+  dom::WorkerPrivate* workerPrivate = dom::GetCurrentThreadWorkerPrivate();
+  bool usePlaceholder =
+      nsContentUtils::ShouldResistFingerprinting(workerPrivate->GetPrincipal());
   CanvasRenderingContextHelper::ToBlob(aCx, global, callback, aType, aParams,
                                        usePlaceholder, aRv);
 